### PR TITLE
feat: cnvkit export wrapper

### PR DESCRIPTION
**Pull request by Smeds**
### Description

wrapper for cnvkit export

### QC

* [x] I confirm that:

For all wrappers added by this PR, 

* there is a test case which covers any introduced changes,
* `input:` and `output:` file paths in the resulting rule can be changed arbitrarily,
* either the wrapper can only use a single core, or the example rule contains a `threads: x` statement with `x` being a reasonable default,
* rule names in the test case are in [snake_case](https://en.wikipedia.org/wiki/Snake_case) and somehow tell what the rule is about or match the tools purpose or name (e.g., `map_reads` for a step that maps reads),
* all `environment.yaml` specifications follow [the respective best practices](https://stackoverflow.com/a/64594513/2352071),
* wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`),
* all fields of the example rules in the `Snakefile`s and their entries are explained via comments (`input:`/`output:`/`params:` etc.),
* `stderr` and/or `stdout` are logged correctly (`log:`), depending on the wrapped tool,
* temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to (see [here](https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir); this also means that using any Python `tempfile` default behavior works),
* the `meta.yaml` contains a link to the documentation of the respective tool or command,
* `Snakefile`s pass the linting (`snakemake --lint`),
* `Snakefile`s are formatted with [snakefmt](https://github.com/snakemake/snakefmt),
* Python wrapper scripts are formatted with [black](https://black.readthedocs.io).
* Conda environments use a minimal amount of channels, in recommended ordering. E.g. for bioconda, use (conda-forge, bioconda, nodefaults, as conda-forge should have highest priority and defaults channels are usually not needed because most packages are in conda-forge nowadays).

### CODE REVIEW

File: bio/cnvkit/export/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: "cnvkit export"
+description: >
+  Convert copy number ratio tables (.cnr files) or segments (.cns) to another format..
+url: https://cnvkit.readthedocs.io/en/stable/pipeline.html
```
Comment (Smeds) [inline bio/cnvkit/export/meta.yaml:4]:
```suggestion
url: https://cnvkit.readthedocs.io/en/stable/importexport.html#export
```

File: bio/cnvkit/export/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name: "cnvkit export"
+description: >
+  Convert copy number ratio tables (.cnr files) or segments (.cns) to another format..
```
Comment (fgvieira) [inline bio/cnvkit/export/meta.yaml:3]:
```suggestion
  Convert copy number ratio tables (.cnr files) or segments (.cns) to another format.
```

File: bio/cnvkit/export/wrapper.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+__license__ = "MIT"
+
+from snakemake.shell import shell
+from os.path import splitext
+
+log = snakemake.log_fmt_shell(stdout=False, stderr=True)
+
+extra = snakemake.params.get("extra", "")
+
+file_extension = splitext(snakemake.output[0])[1][1:]
```
Comment (fgvieira) [inline bio/cnvkit/export/wrapper.py:13]: Would this also work with a `vcf.gz`?

File: test.py [hunk 4]
```diff
@@ -1180,6 +1180,49 @@ def test_bwa_mapping_meta():
         ],
     )
 
+@skip_if_not_modified
+def test_cnvkit_target():
```
Comment (fgvieira) [inline test.py:1184]:
```suggestion
def test_cnvkit_export():
```

File: bio/cnvkit/export/wrapper.py (hunk 3 above)
Comment (Smeds) [reply bio/cnvkit/export/wrapper.py:13]: I will do a small test to see if the software can output a compressed vcf

Comment (Smeds) [reply bio/cnvkit/export/wrapper.py]: The tool didn't support vcf.gz as output! But I have update the wrapper to support ".gz" as output.

File: bio/cnvkit/export/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+name: "cnvkit export"
+description: >
+  Convert copy number ratio tables (.cnr files) or segments (.cns) to another format.
+url: https://cnvkit.readthedocs.io/en/stable/importexport.html#export
+input:
+  - cns: cns or cnr file(s)
+output:
+  - bed/vcf/cdt/seq file
```
Comment (fgvieira) [inline bio/cnvkit/export/meta.yaml:8]:
```suggestion
  - bed/vcf/vcf.gz/cdt/seq file
```

**Review by fgvieira — APPROVED** [review]

**Pull request merged**